pkg/lvgl/contrib/lvgl.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2019 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     pkg_lvgl
+ * @{
+ *
+ * @file
+ * @brief       LittlevGL glue code
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ * @}
+ */
+
+#include "thread.h"
+#include "mutex.h"
+
+#include "xtimer.h"
+#include "log.h"
+
+#include "lvgl/lvgl.h"
+#include "lv_conf.h"
+#include "lvgl_riot.h"
+#include "disp_dev.h"
+
+#ifndef LVGL_THREAD_PRIO
+#define LVGL_THREAD_PRIO        (THREAD_PRIORITY_MAIN + 1)
+#endif
+
+#ifndef LVGL_COLOR_BUF_SIZE
+#define LVGL_COLOR_BUF_SIZE     (LV_HOR_RES_MAX * 5)
+#endif
+
+static char _task_thread_stack[THREAD_STACKSIZE_MAIN];
+
+static lv_disp_buf_t disp_buf;
+static lv_color_t buf[LVGL_COLOR_BUF_SIZE];
+static disp_dev_t *_dev = NULL;
+
+void *_task_thread(void *arg)
+{
+    (void) arg;
+
+    while (1) {
+        lv_task_handler();
+        xtimer_usleep(5 * US_PER_MS);
```

Comment:
does this need to be called 200 times per second?

Reply:
Seems so: https://docs.littlevgl.com/en/html/porting/task-handler.html

Reply:
There's ways around: https://docs.littlevgl.com/en/html/porting/sleep.html
(we can improve that later)

Reply:
Thanks, I'll give this a try.

Reply:
> we can improve that later

I can't think of an easy solution for this or without having to patch littlevgl. Maybe I'm missing something.